dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -14,6 +14,8 @@ def all
     if @user.persisted?
       flash.notice = I18n.t('auth.signed_in')
       sign_in_and_redirect @user
+    elsif User.find_by_email_or_hashed_email(@user.email)
+      redirect_to "/users/sign_in?providerNotLinked=#{@user.provider}&email=#{@user.email}"
```

Comment:
Importantly, the `User.from_omniauth` populates `User.email`. Normally, a `User` would not (necessarily) have this field populated, as we wipe plaintext emails for students.